### ISSUE TITLE
format error (either latex or html) with aside in example statement

### ISSUE THREAD

**Issue by sepvar** [PreTeXtBook/pretext#746]
From discussion (point 3) https://groups.google.com/forum/?utm_source=digest&utm_medium=email#!topic/pretext-support/c6feirQqC-c

I created a minimal example that shows the problem.  

Summary: I would like an aside as the first item in the statement of an example.  This screws up the LaTeX format.  Modifications to fix it either cause a different format problem in LaTeX or in HTML.
[minimal-example.txt](https://github.com/rbeezer/mathbook/files/1349243/minimal-example.txt)

Joe

**Comment by sepvar** [comment]
Follow-up.  See discussion from Oct 2, 2017 in https://github.com/rbeezer/mathbook/issues/320

When I use the thmbox package, this issue is fixed for those environments that are redefined in the thmbox style.  If I use the thmbox package but do not specify that specific environment then the issue above continues to be an issue for that environment.  (So, for example, I can use the package to fix example without fixing heuristic.)

**Comment by rbeezer** [comment]
I don't believe an aside is meant to go in an example.  An aside is more like a block, a peer of an example.

**Comment by jjrsylvestre** [comment]
Oh OK. I've tended to put aside in example when the content of the aside is relevant only to the example. I've also done

````
activity
  task
    p
    aside
````

and there isn't the same issue of the aside being pushed off to the side.

**Comment by jjrsylvestre** [comment]
Should the schema reflect this then? Right now

````
            ExampleLike =
                MetaDataTitleOptional,
                (
                    (BlockStatement)+ |
                    (Statement, Hint*, Answer*, Solution*) |
                    (IntroductionStatement?, Task+, ConclusionStatement?)
                )
````
and
````
            BlockStatement =
                BlockText |
                Figure | Aside |
                SideBySide | SideBySideGroup | Sage
````
Where else should aside not go?

**Comment by rbeezer** [comment]
Well, my memory was faulty, I guess. :-(

I'll investigate further next week.

**Comment by jjrsylvestre** [comment]
I don't know if this is related to the aside issue in this thread, but has similar result:

Sometimes something like
````
example
    p
    figure
    p
````
causes the figure to be on the same line as the last line of the preceding p, so that the figure is pushed off toward the right margin instead of appearing between the paragraphs.

To fix it I have my post-processing routines insert a hard linebreak before the figure.

Don't know if this is related to
https://groups.google.com/g/pretext-dev/c/eNBKkaD2Gbk/m/T_4X2wRmAAAJ
in any way.

**Comment by rbeezer** [comment]
I had not forgotten this one!  I think this is exactly what is going on in the recent thread you reference.  And I **just** pushed a group of changes that includes "fixing" asides.

Do a pull, or wait from tomorrow's CLI nightly, and then can you test some situations with your hard-break fixes removed?  Thanks.

**Comment by rbeezer** [comment]
I misspoke.  I'll need to fix examples to correct this.  More later, gotta run right now.

**Comment by jjrsylvestre** [comment]
Appears to be fixed by ed6467a5 (both the aside issue and the figure issue).
